oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -420,6 +428,60 @@ EnvironmentProvider getEnvironmentProvider() {
     return environmentProvider;
   }
 
+  static void validateTokenUrl(String tokenUrl) {
+    List<Pattern> patterns = new ArrayList<>();
+    patterns.add(Pattern.compile("^[^\\.\\s\\/\\\\]+\\.sts\\.googleapis\\.com$"));
+    patterns.add(Pattern.compile("^sts\\.googleapis\\.com$"));
+    patterns.add(Pattern.compile("^sts\\.[^\\.\\s\\/\\\\]+\\.googleapis\\.com$"));
+    patterns.add(Pattern.compile("^[^\\.\\s\\/\\\\]+\\-sts\\.googleapis\\.com$"));
```

Comment:
Any chance at simplifying these?

Reply:
I reviewed the tests and am happy with the coverage, but still consider if there is a simpler way to express this?

Reply:
I'd prefer to keep it like this rather than have one long complicated regex.